source/nvwave.py

```diff
@@ -286,7 +300,13 @@ def _idleUnbuffered(self):
 				if not self._waveout:
 					return
 				if self.closeWhenIdle:
-					self._close()
+					if not self._closeTimer:
+						# We need a cancellable timer, so we can't use core.callLater.
+						self._closeTimer = wx.PyTimer(self._close)

```

Comment:
Why not use here a regular `threading.Timer` or even `wx.Timer`?
Note `wx.PyTimer` seems to have gone away of wxPython documentation as of the Phoenix migration for some reason (maybe a bad one, no clue)
Sorry if I'm missing the point here.

Reply:
threading.Timer would spawn another thread and execute the function there. That's pretty wasteful, since we already have an event loop in our main thread. It also increases the potential for race conditions.
wx.Timer doesn't allow you to pass in a callback, so you have to subclass it. That makes things a lot less readable.